core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -735,6 +736,43 @@ public long reinitializeFile(LockedInodePath inodePath, long blockSizeBytes, lon
     return file.getId();
   }
 
+  /**
+   * Locks a specific descendant of a particular {@link LockedInodePath}. It does not extend the
+   * {@link LockedInodePath}, it only locks the descendant.
+   *
+   * @param inodePath the root to start locking
+   * @param lockMode the lock type to use
+   * @param descendantUri the path to the descendant that we are locking
+   * @return  an {@link InodeLockList} representing the list of descendants that got locked as
+   * a result of this call.
+   * @throws FileDoesNotExistException if inode does not exist
+   */
+  public InodeLockList lockDescendant(LockedInodePath inodePath, LockMode lockMode,
+      AlluxioURI descendantUri) throws FileDoesNotExistException {
+    InodeLockList inodeGroup = new InodeLockList();
+    // Check if the descendant is really the descendant of inodePath
+    try {
+      if (!PathUtils.hasPrefix(descendantUri.getPath(), inodePath.getUri().getPath())
+          || descendantUri.getPath().equals(inodePath.getUri().getPath())) {
```

Comment:
If the descendant is same as the input path, what should happen? Is this also an error case?

Can you add a test case for this?

Reply:
done